src/journal/journald-syslog.c

```diff
@@ -304,44 +304,55 @@ void server_process_syslog_message(
                 size_t label_len) {
 
         char syslog_priority[sizeof("PRIORITY=") + DECIMAL_STR_MAX(int)],
-             syslog_facility[sizeof("SYSLOG_FACILITY=") + DECIMAL_STR_MAX(int)], *msg;
-        const char *message = NULL, *syslog_identifier = NULL, *syslog_pid = NULL;
+             syslog_facility[sizeof("SYSLOG_FACILITY=") + DECIMAL_STR_MAX(int)];
+        const char *message = NULL, *syslog_identifier = NULL, *syslog_pid = NULL, *msg;
         _cleanup_free_ char *identifier = NULL, *pid = NULL;
         int priority = LOG_USER | LOG_INFO, r;
         ClientContext *context = NULL;
         struct iovec *iovec;
-        size_t n = 0, m, i;
+        size_t n = 0, m, i, leading_ws;
 
         assert(s);
         assert(buf);
+        assert(buf_len > 0);
+        /* We know that buf is non-empty. It is also always NUL-terminated and can be used a string */
```

Comment:
Hmm, these are weird calling conventions... passing a size *AND* requiring a NUL byte at the end is pretty uncommon. If we do this, then I think this deserves a bigger comment really, and it should clarify that the trailing NUL byte is there, but not counted in `buf_len`.

It might also be worth making these assumptions explicit, by adding `assert(buf[buf_len] == 0` or so.

Also, I think the name "buf_len" is a bit misleading now, given that the buffer is one byte larger than buf_len now. Maybe rename this to "raw" and "raw_len" or so?

Reply:
I renamed buf_len to raw_len, and added a comment. This terminating byte allows strspn to be used, so I think it's worth keeping.